### PR TITLE
Add TimerTask

### PR DESCRIPTION
**Pull request by davidor**
This PR adds a new module that allows running periodic tasks that can be canceled.

### CODE REVIEW

File: gateway/src/resty/concurrent/timer_task.lua [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  self.args = options.args
+  self.interval = options.interval or default_interval_seconds
+  self.id = id
+
+  _M.register_task(id)
+
+  return self
+end
+
+local run_periodic, schedule_next, timer_execute
```
Comment (mikz) [inline gateway/src/resty/concurrent/timer_task.lua]: That is very weird style. Why not just define each function as local?

Comment (mikz) [reply gateway/src/resty/concurrent/timer_task.lua]: I see. They all depend on each other so they need to be defined this way.

File: spec/resty/concurrent/timer_task_spec.lua [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    local interval = 10
+
+    before_each(function()
+      ngx_timer_stub = stub(ngx_timer, 'at')
+    end)
+
+    describe('when the task is active', function()
+      it('runs the task', function()
+        local timer_task = TimerTask.new(func, { args = args, interval = interval })
+        local func_stub = stub(timer_task, 'task')
```
[the remark this answers is not in the document]
Comment (mikz) [reply spec/resty/concurrent/timer_task_spec.lua]: Codeclimate does not update comments after initial review?

File: gateway/src/resty/concurrent/timer_task.lua (hunk 1 above)
Comment (davidor) [reply gateway/src/resty/concurrent/timer_task.lua]: Exactly.

File: gateway/src/resty/concurrent/timer_task.lua [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+run_periodic = function(self)
+  if not _M.task_is_active(self.id) then return end
+
+  self.task(unpack(self.args))
```
Comment (mikz) [inline gateway/src/resty/concurrent/timer_task.lua]: Would be good to use https://github.com/3scale/apicast/pull/783/commits/4a5a8159602c5e1173e0a6dabb6529ee34f10551 to make this safe.

File: spec/resty/concurrent/timer_task_spec.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidor) [reply spec/resty/concurrent/timer_task_spec.lua]: Looks like it writes another comment without deleting the old one.

File: gateway/src/resty/concurrent/timer_task.lua (hunk 3 above)
Comment (davidor) [reply gateway/src/resty/concurrent/timer_task.lua]: Yes, I can do that after resty.concurrent is merged.

**Pull request merged**